guide/src/function.md

```diff
@@ -233,3 +249,41 @@ fn module_with_fn(py: Python, m: &PyModule) -> PyResult<()> {
 
 # fn main() {}
 ```
+
+## Accessing the FFI functions
+
+In order to make Rust functions callable from Python, PyO3 generates a 
+`extern "C" Fn(slf: *mut PyObject, args: *mut PyObject, kwargs: *mut PyObject) -> *mut Pyobject`
+function and embeds the call to the Rust function inside this FFI-wrapper function. This
+wrapper handles extraction of the regular arguments and the keyword arguments from the input
+`PyObjects`. Since this function is not user-defined but required to build a `PyCFunction`, PyO3
+offers the `raw_pycfunction!()` macro to get the identifier of this generated wrapper.
+
+E.g.:
+
+```rust
+use pyo3::prelude::*;
+use pyo3::types::PyCFunction;
+use pyo3::raw_pycfunction;
+
+#[pyfunction]
+fn some_fun() -> PyResult<()> {
+    Ok(())
+}
+
+#[pymodule]
+fn module(_py: Python, module: &PyModule) -> PyResult<()> {
+    let ffi_wrapper_fun = raw_pycfunction!(some_fun);
+    let docs = "Some documentation string with null-termination\0";
+    let py_cfunction =
+        PyCFunction::new_with_keywords(ffi_wrapper_fun, "function_name", docs, module.into())?;
+    module.add_function(py_cfunction)
+}
+
+# fn main() {}
+```
+
+The `wrap_pyfunction` macro can be used to directly get a `PyCFunction` given a
+`#[pyfunction]` and a `PyModule`: `wrap_pyfunction!(rust_fun, module)`. In the single-argument
+variant, the macro only takes the identifier of the Rust function and returns a function that
+can be called with either `Python` or `PyModule` and returns a `PyCFunction`.
```

Comment:
I think this paragraph may go into more detail than needed. (I'd rather not talk about the single-argument variant of `wrap_pyfunction` any more so that the ecosystem naturally migrates to the two-argument form.)